app/Http/Controllers/Table/PortNacController.php

```diff
@@ -72,10 +73,14 @@ protected function sortFields($request)
      */
     public function baseQuery($request)
     {
-        return PortsNac::select('port_id', 'mac_address', 'ip_address', 'vlan', 'domain', 'host_mode', 'username', 'authz_by', 'timeout', 'time_elapsed', 'time_left', 'authc_status', 'authz_status', 'method')
-            ->where('device_id', $request->device_id)
-            ->hasAccess($request->user())
-            ->with('port');
+        $query = PortsNac::select('device_id', 'port_id', 'mac_address', 'ip_address', 'vlan', 'domain', 'host_mode', 'username', 'authz_by', 'timeout', 'time_elapsed', 'time_left', 'authc_status', 'authz_status', 'method')
+                ->hasAccess($request->user())
+                ->with('port');
+        $query->when(is_numeric($request->device_id), function ($q) use ($request) {
+            return $q->where('device_id', $request->device_id);
+        });
+
+        return $query;
```

Comment:
```suggestion
        return PortsNac::select('device_id', 'port_id', 'mac_address', 'ip_address', 'vlan', 'domain', 'host_mode', 'username', 'authz_by', 'timeout', 'time_elapsed', 'time_left', 'authc_status', 'authz_status', 'method')
            ->when($request->device_id, fn($q, $id) => $q->where('device_id', $id))
            ->hasAccess($request->user())
            ->with('port');
```

Just a little cleaner

Reply:
on its way